qiskit/primitives/base_estimator.py

```diff
@@ -227,12 +258,94 @@ def __call__(
             values = parameter_values[i].
 
         Args:
-            circuit_indices: the list of circuit indices.
-            observable_indices: the list of observable indices.
+            circuits: the list of circuit indices.
+            observables: the list of observable indices.
             parameter_values: concrete parameters to be bound.
             run_options: runtime options used for circuit execution.
 
         Returns:
             EstimatorResult: The result of the estimator.
+
+        Raises:
+            QiskitError: For mismatch of object id.
+            QiskitError: For mismatch of length of Sequence.
         """
+
+        # Support ndarray
+        if isinstance(parameter_values, np.ndarray):
+            parameter_values = parameter_values.tolist()
+
+        # Allow optional
+        if parameter_values is None:
+            for i in circuits:
+                if len(self._circuits[i].parameters) != 0:
+                    raise QiskitError(
+                        f"The {i}-th circuit ({len(circuits)}) is parameterised,"
```

Comment:
Just a minor question, why does this include the number of circuits to be executed in the warning? 🙂

Reply:
Is it because it is more convenient to know how many of them there are?
@rathishcholarajan Do you have any idea?

Reply:
Not sure either, it was in the internal implementation. We can remove it if it doesn't make sense. Good catch @Cryoris!
```suggestion
                        f"The {i}-th circuit is parameterised,"
```